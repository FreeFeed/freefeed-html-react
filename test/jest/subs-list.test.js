/* global jest, describe, it, expect */
import { render } from '@testing-library/react';
import * as reactRedux from 'react-redux';

import SubsList from '../../src/components/subs-list';

const defaultState = {
  user: {
    frontendPreferences: {},
  },
};

describe('SubsList', () => {
  it('Renders a list of subscribers', () => {
    const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
    useSelectorMock.mockImplementation((selector) => selector(defaultState));

    const { asFragment } = render(
      <SubsList
        title="Subscribers"
        listSections={[
          {
            title: 'Section',
            users: [
              {
                id: 'u1',
                screenName: 'User 1',
                username: 'user1',
                isMutual: true,
                profilePictureUrl: 'pic1.jpg',
              },
              {
                id: 'u2',
                screenName: 'User 2',
                username: 'user2',
                isMutual: false,
                profilePictureUrl: 'pic2.jpg',
              },
            ],
          },
        ]}
        showSectionsTitles
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
