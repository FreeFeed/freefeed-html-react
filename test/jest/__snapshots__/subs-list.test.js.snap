// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubsList Renders a list of subscribers 1`] = `
<DocumentFragment>
  <div>
    <h3>
      <span>
        Subscribers
      </span>
    </h3>
    <h4
      class="tile-list-subheader"
    >
      Section
    </h4>
    <div>
      <ul
        class="tile-list with-actions"
      >
        <li>
          <div
            class="avatar"
          >
            <a
              class="picture"
            >
              <img
                alt="Profile picture of user1"
                height="50"
                src="pic1.jpg"
                width="50"
              />
            </a>
          </div>
          <span
            class="user-name-wrapper"
          >
            <a
              class=""
            >
              <span
                dir="auto"
              >
                User 1
              </span>
            </a>
          </span>
          <div
            class="user-ismutual"
          >
            mutual
          </div>
        </li>
        <li>
          <div
            class="avatar"
          >
            <a
              class="picture"
            >
              <img
                alt="Profile picture of user2"
                height="50"
                src="pic2.jpg"
                width="50"
              />
            </a>
          </div>
          <span
            class="user-name-wrapper"
          >
            <a
              class=""
            >
              <span
                dir="auto"
              >
                User 2
              </span>
            </a>
          </span>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
